packages/eslint-plugin/src/rules/no-unnecessary-type-constraint.ts

```diff
@@ -89,6 +88,9 @@ export default util.createRule({
           suggest: [
             {
               messageId: 'removeUnnecessaryConstraint',
+              data: {
+                constraint,
```

Comment:
This is an actual bug. This data was missing for the suggestion message and would thus show as `{{constraint}}` to the user instead of the actual value. I updated the tests to test this data.